### DISCUSSION TITLE
Frontend<->Backend communication in `production` mode

### DISCUSSION THREAD

**Discussion by ctmbl** [iScsc/iscsc.fr#23] [Show and tell]
### Introduction
#20 introduced the Docker system needed to deploy the iScsc website and thus introduce separation between development and production modes.
In this [thread](https://github.com/iScsc/iscsc.fr/pull/20#discussion_r1006703930) we discussed about the differences between dev and prod modes, especially everything related to the "proxy" needed for communication between the backend and the frontend.

### Context of the discussion
To summarize it I asked why does the routes are different in dev and prod mode and @atxr [answered](https://github.com/iScsc/iscsc.fr/pull/20#discussion_r1006703930) that thanks to a "proxy" in **dev mode** the frontend can easily contact the backend as if it where a local URL, without taking into account the network _as I understand it_. 
However in **prod** mode this "proxy" apparently doesn't work because of some react build that shrink the features and capabilities to make the website lighter and faster for deployment.  
Thus, a "reverse proxy" is used to replace it, it uses `nginx` [as originally configured here](https://github.com/atxr/iscsc.fr/blob/v0.2.0/production-mode/nginx/nginx.conf.template) and its role is simply to ease communication between frontend and backend, dealing with URL, ports and so on...

Now it's time for @atxr to correct what I just summarized and add details I ignore as well as good documentation/video about these topics and technologies!

**Comment by dolphinau** [comment]
I'll do a little summary tonight!

**Reply by ctmbl** [reply]
tks!

**Reply by dolphinau** [reply]
Actually your summary is very good! Do you still have some question after the last presentation?

**Reply by ctmbl** [reply]
I quite like your explication of what `nginx` is if you can write it down it would be great!
But I didn't get what CORS is?
